src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -667,33 +507,45 @@ def resolve_analyzed_atomic_cell(
             (Optional[AtomicCell]): The resolved `AtomicCell` section or None if the cell_type
             is not recognized.
         """
-        if cell_type not in ['primitive', 'conventional']:
-            logger.error(
-                "Cell type not recognized, only 'primitive' and 'conventional' are allowed."
-            )
+        # Define a mapping for each supported cell type
+        cell_type_map = {
+            'primitive': {
+                'wyckoff': symmetry_analyzer.get_wyckoff_letters_primitive,
+                'equivalent': symmetry_analyzer.get_equivalent_atoms_primitive,
+                'system': symmetry_analyzer.get_primitive_system,
+            },
+            'conventional': {
+                'wyckoff': symmetry_analyzer.get_wyckoff_letters_conventional,
+                'equivalent': symmetry_analyzer.get_equivalent_atoms_conventional,
+                'system': symmetry_analyzer.get_conventional_system,
+            },
+        }
+
+        mapping = cell_type_map.get(cell_type)
+        if mapping is None:
+            logger.error(f'Cell type {cell_type} is not supported.')
+            return None
+
+        try:
+            wyckoff = mapping['wyckoff']()
+            equivalent_atoms = mapping['equivalent']()
+            system = mapping['system']()
+        except Exception as e:
+            logger.error('Error extracting symmetry data', exc_info=e)
             return None
-        wyckoff = getattr(symmetry_analyzer, f'get_wyckoff_letters_{cell_type}')()
-        equivalent_atoms = getattr(
-            symmetry_analyzer, f'get_equivalent_atoms_{cell_type}'
-        )()
-        system = getattr(symmetry_analyzer, f'get_{cell_type}_system')()
 
-        positions = system.get_scaled_positions()
         cell = system.get_cell()
-        atomic_numbers = system.get_atomic_numbers()
-        labels = system.get_chemical_symbols()
 
-        atomic_cell = AtomicCell(type=cell_type)
+        # Create the cell (or atomic cell) for geometry only
+        atomic_cell = Cell(type=cell_type)
         atomic_cell.lattice_vectors = cell * ureg.angstrom
-        for label, atomic_number in zip(labels, atomic_numbers):
-            atom_state = AtomsState(chemical_symbol=label, atomic_number=atomic_number)
-            atomic_cell.m_add_sub_section(AtomicCell.atoms_state, atom_state)
-        atomic_cell.positions = (
-            positions * ureg.angstrom
-        )  # ? why do we need to pass units
+        # Note: positions are no longer stored in the cell in the new design.
```

Comment:
I like the the extra comments in general, some suggestions: 
1. use a "#!" for more visibility
2. clarify where the positions are "# ! Positions are now stored directly under `model_system`"
3. clarify what new design means (you can use the new release planned): "# ! Positions are stored directly under `model_system` in nomad-simulations>=0.4"